### ISSUE TITLE
[cloud-provider-aws] Spot Management Improvements

### ISSUE THREAD

**Issue by zuzzas** [deckhouse/deckhouse#4019]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Currently, spot instances are in a weird way. They are not managed 

AWS [Fleet](https://docs.aws.amazon.com/systems-manager/latest/userguide/fleet.html) is a wonderful way to manage "a fleet" of instances. You can specify desired total size, availability requirements, and other considerations, and it manages itself by ordering available instance of different sizes. We can insert a cloud-config script that is generated by node-manager to bootstrap these Nodes into a cluster. Fleet will delete them automatically, and CCM will then delete Nodes from the cluster.

### Proposed Solution

Write a guide on how to use cloud-config generated by node-manager module and use it to bootstrap AWS Fleet instances.

**Comment by nabokihms** [comment]
https://cluster-api-aws.sigs.k8s.io/topics/spot-instances.html

**Comment by zuzzas** [comment]
Indeed, but it requires ClusterAPI and will still be inferior to AWS Fleet. Or not.

**Comment by raabdullaev-fl** [comment]
There are already Cluster API provider in the Deckhouse. In our future plans we would like to migrate from MCM-based AWS provider to Cluster API and as a result add AWS Fleet support.

**Comment by raabdullaev** [comment]
Moving to Carpenter is considered.
